cpp/src/DataStorm/ConnectionManager.cpp

```diff
@@ -60,27 +60,20 @@ ConnectionManager::remove(const Ice::ConnectionPtr& connection)
         connection->setCloseCallback(nullptr);
         _connections.erase(p);
     }
+
     exception_ptr ex;
     try
     {
-        [[maybe_unused]] auto _ = connection->getInfo();
+        connection->throwException();
```

Comment:
It is not clear why we were using `getInfo` to retrieve the connection exception. Using throwException seems the more correct way to do that.

Reply:
I would add an `assert(false);` after this call just in case it doesn't throw.

Reply:
I would add `[[noreturn]]` to throwException.

Reply:
I don't think `throwException` will throw if there's no exception. Like when the connection is still alive.

Reply:
> I would add [[noreturn]] to throwException.

Isn't this only for functions that never return. This function can return normally if the connection is still alive.

Reply:
> I would add an assert(false); after this call as we expect it to always throw.

I don't think we should. ConnectionManager::remove can also be called from SessionI::destroyImpl and then it is not warantee that the connection would throw.